### DISCUSSION TITLE
symbols-outline integration with zk for backlinks

### DISCUSSION THREAD

**Discussion by kha-dinh** [zk-org/zk-nvim#134] [Ideas]
Hi, I have had a great time with this plugin and zk. However, I miss the obsidian feature where you can visually see all the backlinks to the current note.

Integrating such a feature with simrat39's [symbols-outline.nvim](https://github.com/simrat39/symbols-outline.nvim) seems like a good idea, so I went ahead and implemented it in [my fork](https://github.com/kha-dinh/symbols-outline.nvim). For now, it can show the list of backlinks plus the mentioned line. Also, you can visit the note by selecting the backlink.
The next step is turning it into a separated dependent plugin of symbol-outline. I hope someone find this helpful.

Here is an example:
<img width="1439" alt="Screenshot 2023-05-27 at 19 59 49" src="https://github.com/mickael-menu/zk-nvim/assets/16218516/8e5d21aa-32a6-4919-8d62-3a32b2085614">

**Comment by mickael-menu** [comment]
Thanks for sharing, this looks pretty good!

**Comment by tjex** [comment]
@kha-dinh amazing! I've also been wanting this. @kabouzeid , what do you think? 

Kha-dinh, are you up for making a PR? Even if there are still some things to iron out, it would be great to integrate the work you've put into it.

**Reply by kha-dinh** [reply]
Hi @tjex , Since the time I posted this issue, I have stopped using zk as a daily driver.

However, as development seems to be picking up, I would be happy to get back into it and contribute this feature.

One thing I found during implementation of this is that `symbol-outline` did not cross file referencing, so I had to make some hacks to `symbol-outline` to support it.

**Reply by kha-dinh** [reply]
Going forward, we can either maintain a fork of `symbol-outline`, implement our own outliner, or choose a better alternative.

**Reply by tjex** [reply]
ahh I overlooked that symbol-outline was in fact a third party plugin. I thought it was some pretty inbuilt quickfix list! my bad.

Integrating and managing another dependancy isn't on the cards for the moment I would say (particularly as the symbol-outline project is archived as of Jan).

If we consider looking at this again, we can ping you. I'll transfer this discussion to the ideas section as well in case it gains some traction there in the meantime.

**Reply by karimknaebel** [reply]
We should definitely support both document symbols and workspace symbols LSP APIs. However, for document symbols, I don't think backlinks makes sense here, it should be only forward links from the current note. Workspace symbols could just be a list of every single note in the Zettelkasten. Especially workspace symbols makes `zk` much more usable in other editors such as VSCode that don't have a special `zk` plugin.
